src/core/qgsdatasourceuri.h

```diff
@@ -168,6 +168,8 @@ class CORE_EXPORT QgsDataSourceURI
     QString database() const;
     //! Returns the port
     QString port() const;
+    //! Returns the driver
+    QString driver() const;
```

Comment:
Added a setter, although similar values like port don't have an explicit setter.
Also updated qgsdatasourceuri.sip.
Added corresponding test in testqgsdatasourceuri.cpp although it isn't clear how this works.